pkg/filter/proxy/pool.go

```diff
@@ -127,11 +129,30 @@ func (p *pool) status() *PoolStatus {
 	return s
 }
 
+var requestPool = sync.Pool{
+	New: func() interface{} {
+		return &request{}
+	},
+}
+
+var httpstatResultPool = sync.Pool{
+	New: func() interface{} {
+		return &gohttpstat.Result{}
+	},
+}
+
 func (p *pool) handle(ctx context.HTTPContext, reqBody io.Reader, client *http.Client) string {
-	addTag := func(subPrefix, msg string) {
-		tag := stringtool.Cat(p.tagPrefix, "#", subPrefix, ": ", msg)
+	addLazyTag := func(subPrefix, msg string, intMsg int) {
```

Comment:
Why not just convert int into string outside `addLazyTag`. The extra message parameter would be a little confusing.

Reply:
Under very heavy load it takes a lot of time to convert integers to string, so idea here is to push the Itoa to logging, HTTPContext.OnFinish. Then when logging is disabled (for example when running stress tests), expensive string concatenations and Itoa are not executed and the performance increases slightly. 
But I agree that it's confusing, I will add an extra comment for that